### ISSUE TITLE
MDM detection

### ISSUE THREAD

**Issue by tudorcraciun** [doronz88/pymobiledevice3#1073]
Hello! 
I have just stumbled upon this tool, and I was wondering if you found any way to detect if an MDM/DEP lock is present on a device without any configuration? A query to _com.apple.mobile.chaperone_ with the key _DeviceIsChaperoned_ returns true only after some setup is made on the device, and I cannot find a way to work around that.

**Comment by doronz88** [comment]
I guess you are trying to get the value while in the `Setup.app`?

**Comment by tudorcraciun** [comment]
If by Setup.app you mean the configuration wizard where you go through selecting Language, Country, etc, then yes. The ideal scenario would be to check for the MDM/DEP lock on a freshly reset device.

**Comment by AnNEDoMini** [comment]
You can make activation, then query DeviceIsChaperoned, after that you can reset it to bring to blank state. If I understand you correctly

**Comment by tudorcraciun** [comment]
Yes, but activation is not enough by itself, some more setup is required after that point.
